src/Assignments/ProblemDetails.tsx

```diff
@@ -20,17 +20,40 @@ export const ProblemDetails: React.FC<ProblemDetailsProps> = ({
     topic,
     attemptsRemaining,
 }) => {
-    const isAssessment = !_.isNil(topic?.topicAssessmentInfo?.studentTopicAssessmentInfo?.[0]);
-    const isClosed = isAssessment && topic?.topicAssessmentInfo?.studentTopicAssessmentInfo?.[0].isClosed;
-    const versionStartTime = topic?.topicAssessmentInfo?.studentTopicAssessmentInfo?.[0].startTime;
-    const versionEndTime = topic?.topicAssessmentInfo?.studentTopicAssessmentInfo?.[0].endTime;
+    if (_.isNil(topic)) {
+        console.error('Problem details requested without a topic');
+    }
+
+    let version: StudentTopicAssessmentFields | undefined;
+
+    if (
+        !_.isNil(topic) &&
+        !_.isNil(topic.topicAssessmentInfo) &&
+        !_.isEmpty(topic?.topicAssessmentInfo?.studentTopicAssessmentInfo)
+    ) {
+        version = _.maxBy(topic.topicAssessmentInfo.studentTopicAssessmentInfo, 'startTime');
+        if (_.isNil(version)) {
+            console.error('We have versions, but an attempt to set the current version failed.');
+        }
+    }
+
+    const isVersionedAssessment = (topic?.topicTypeId === 2 && !_.isNil(version));
```

Comment:
as @PanopticaRising pointed out we have an enum for this on the frontend, but I can fix this when I fix mine

Reply:
I couldn't get the frontend ENUM to work for me - happy to receive a tutorial